app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -212,21 +159,28 @@ private fun ConversationScreenContent(
     onSendAttachment: (AttachmentBundle?) -> Unit,
     onDownloadAsset: (String) -> Unit,
     onImageFullScreenMode: (String) -> Unit,
-    onMessageComposerError: (ConversationErrors) -> Unit,
+    onMessageComposerError: (ConversationSnackbarMessages) -> Unit,
     conversationState: ConversationViewState,
     conversationScreenState: ConversationScreenState
 ) {
     val lazyListState = rememberLazyListState()
     val coroutineScope = rememberCoroutineScope()
+    val context = LocalContext.current
 
     LaunchedEffect(conversationState.messages) {
         lazyListState.animateScrollToItem(0)
     }
 
-    conversationState.onError?.let { errorCode ->
-        val errorMessage = getErrorMessage(errorCode)
-        LaunchedEffect(conversationState.onError) {
-            conversationScreenState.snackBarHostState.showSnackbar(errorMessage)
+    conversationState.onSnackbarMessage?.let { msgCode ->
+        val (message, actionLabel) = getSnackbarMessage(msgCode)
+        LaunchedEffect(conversationState.onSnackbarMessage) {
+            val snackbarResult = conversationScreenState.snackBarHostState.showSnackbar(message = message, actionLabel = actionLabel)
+            when {
+                // Show downloads folder when clicking on Snackbar cta button
+                msgCode is OnFileDownloaded && snackbarResult == SnackbarResult.ActionPerformed -> {
+                    context.startActivity(Intent(DownloadManager.ACTION_VIEW_DOWNLOADS))
+                }
+            }
```

Comment:
minor thing, but maybe this could be moved to conversationScreenState ?

Reply:
I don't think there should be any logic on conversationScreenState. That should only be a holder of information. And this logic could not be transferred to the VM as it is related with Android framework and VM should be agnostic of that. 
I can move that logic to a separate function if you think that makes it more readable?

Reply:
I think ConversationScreenState can have some logic that is related only to to the UI and cannot be done in the ViewModel, but I am fine with private function as well